pkg/cgroups/systemd.go

```diff
@@ -0,0 +1,73 @@
+package cgroups
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	systemdDbus "github.com/coreos/go-systemd/dbus"
+	"github.com/godbus/dbus"
+)
+
+func systemdCreate(path string) error {
+	c, err := systemdDbus.New()
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	slice, name := filepath.Split(path)
+	slice = strings.TrimSuffix(slice, "/")
+
+	var lastError error
+	for i := 0; i < 2; i++ {
+		properties := []systemdDbus.Property{
+			systemdDbus.PropDescription(fmt.Sprintf("cgroup %s", name)),
+			systemdDbus.PropWants(slice),
+		}
+		pMap := map[string]bool{
+			"DefaultDependencies": false,
+			"MemoryAccounting":    true,
+			"CPUAccounting":       true,
+			"BlockIOAccounting":   true,
+		}
+		if i == 0 {
+			pMap["Delegate"] = true
+		}
+		for k, v := range pMap {
+			p := systemdDbus.Property{
+				Name:  k,
+				Value: dbus.MakeVariant(v),
+			}
+			properties = append(properties, p)
+		}
+
+		ch := make(chan string)
+		_, err = c.StartTransientUnit(name, "replace", properties, ch)
+		if err != nil {
+			lastError = err
```

Comment:
If lastErr != nil {
logrus.Errorf(lastErr)
}

Reply:
we return the `lastErr` at the end of the function.  Do we still need to print it here?

Reply:
The problem is if you have multiple errors, then you are only returning the last one.  This will only log the lastError if a new error was found and is about to replace the previous LastError

Reply:
in this case I really want to log only the last error.  `containerd/cgroups` runs a test system  service to check whether `Delegate` is supported.  That makes sense for a daemon as the check is done only once, but with libpod we ended up doing it every time we created a container.

What I do here instead is to first try with `Delegate` enabled and only if that fails, attempt again without setting `Delegate`.  The result is that with recent systemd releases we create only a service, and we run two services only with older versions.  So in the worse case it behaves as before.